src/utils/markets/getMarketsData.ts

```diff
@@ -0,0 +1,123 @@
+  CToken__factory,
+  LendingPool__factory,
+  LendingPoolAddressesProvider__factory,
+  MorphoAaveV2__factory,
+  MorphoAaveV2Lens__factory,
+  MorphoCompound__factory,
+  MorphoCompoundLens__factory,
+  VariableDebtToken__factory,
+} from "@morpho-labs/morpho-ethers-contract";
+import addresses from "@morpho-labs/morpho-ethers-contract/lib/addresses";
+import { MarketMinimal } from "../graph/getGraphMarkets/markets.types";
+import { cFei, pow10BN, WAD } from "../../helpers";
+
+const getMarketsData = async (snapshotBlock: providers.BlockTag, provider: providers.Provider) => {
+  const [compoundParameters, aaveParameters] = await Promise.all([
+    getCompoundMarketsParameters(snapshotBlock, provider),
+    getAaveMarketsParameters(snapshotBlock, provider),
+  ]);
+
+  return {
+    compound: compoundParameters,
+    aave: aaveParameters,
+    markets: [...compoundParameters, ...aaveParameters],
+  };
+};
+const getAaveMarketsParameters = async (snapshotBlock: providers.BlockTag, provider: providers.Provider) => {
+  const overrides = { blockTag: snapshotBlock };
+  const morpho = MorphoAaveV2__factory.connect(addresses.morphoAave.morpho, provider);
+  const lens = MorphoAaveV2Lens__factory.connect(addresses.morphoAave.lens, provider);
+  const lendingPool = LendingPool__factory.connect(await morpho.pool(overrides), provider);
+  const addressesProvider = LendingPoolAddressesProvider__factory.connect(
+    await lendingPool.getAddressesProvider(overrides),
+    provider
+  );
+  const oracle = AavePriceOracle__factory.connect(await addressesProvider.getPriceOracle(overrides), provider);
+
+  const allMarkets = await morpho.getMarketsCreated(overrides);
+
+  return Promise.all(
+    allMarkets.map(async (market) => {
+      const aToken = AToken__factory.connect(market, provider);
+      const [underlying, decimals, totalSupply] = await Promise.all([
+        aToken.UNDERLYING_ASSET_ADDRESS(overrides),
+        aToken.decimals(overrides),
+        aToken.totalSupply(overrides),
+      ]);
+      const { variableDebtTokenAddress } = await lendingPool.getReserveData(underlying, overrides);
+      const debtToken = VariableDebtToken__factory.connect(variableDebtTokenAddress, provider);
+      const [
+        totalBorrow,
+        price,
+        { p2pIndexCursor },
+        { p2pBorrowAmount, poolBorrowAmount, p2pSupplyAmount, poolSupplyAmount },
+      ] = await Promise.all([
+        debtToken.totalSupply(overrides),
+        oracle.getAssetPrice(underlying, overrides),
+        lens.getMarketConfiguration(market, overrides),
+        lens.getMainMarketData(market, overrides),
+      ]);
+      const marketParameters: MarketMinimal = {
+        address: market,
+        totalSupply: totalSupply.mul(price).div(pow10BN(decimals)),
+        totalBorrow: totalBorrow.mul(price).div(pow10BN(decimals)),
+        totalMorphoSupply: p2pSupplyAmount.add(poolSupplyAmount),
+        totalMorphoBorrow: p2pBorrowAmount.add(poolBorrowAmount),
```

Comment:
not in $, like `totalSupply` and `totalBorrow` ?

Reply:
I think that it is easier to reason in underlying token and not in $

Reply:
`totalSupply` and `totalBorrow` was used before to distribute tokens over markets: 
"The lenders (reap. borrowers) of a market in Morpho-Pool receive a share of the amount of MORPHO distributed, which is proportional to the supply (resp. borrow) volume of the corresponding market on Pool."

It was necessary to normalize them in order to distribute over markets. 

For the Age 3, it is not necessary yet to normalize since values are only used to distribute over one market.

I'll change the naming to be less confusing

Reply:
Ok I understand. Now that we no longer need it, I would keep everything is the same (and most simple) unit: underlying.

Reply:
Still needed to recompute the age 2 emissions (retroactively)
I'm not in facor to export the usd conversion since this is not the same computation for aave and compound. Here everything is condensed in the same place. And later, we'll not use anymore USD value except if we need to normalize them in a future age.
Imo the naming can improve the understanding of the returned values

Reply:
With the new naming that's ok !